### DISCUSSION TITLE
Foundry Fixture

### DISCUSSION THREAD

**Discussion by mahabubAlahi** [foundry-rs/foundry#5870] [Q&A] [answered]
Does the foundry have `fixtures` like hardhat's `loadFixture ` to avoid code duplication and reusing common test setups?
https://hardhat.org/tutorial/testing-contracts#reusing-common-test-setups-with-fixtures

**Comment by mds1** [comment, marked as answer]
This is what the `setUp` method does. The resulting state is cached and restored before each test. Please see the docs for more info: https://book.getfoundry.sh/forge/writing-tests
